pkg/federatedtypes/adapter.go

```diff
@@ -30,7 +31,7 @@ import (
 // target resource in member clusters.
 type FederatedTypeAdapter interface {
 	FedClient() fedclientset.Interface
-	Template() FedApiAdapter
+	Template() TemplateAdapter
```

Comment:
Why is this change desirable if `TemplateAdapter` doesn't provide different behavior from `FedApiAdapter`?  `OverrideAdapter` is arguably the bad example here.  Also, this kind of non-functional change would ideally be broken out into a separate commit rather than mixed in with unrelated changes.